extensions/resource-deployment/src/ui/wizardBase.ts

```diff
@@ -0,0 +1,80 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { Disposable } from 'vscode';
+import { WizardPageBase } from './wizardPageBase';
+import { Model } from './model';
+
+export abstract class WizardBase<T, M extends Model> {
+	private customButtons: azdata.window.Button[] = [];
+	private pages: WizardPageBase<T>[] = [];
+
+	public wizardObject: azdata.window.Wizard;
+	public toDispose: Disposable[] = [];
+	public get model(): M {
```

Comment:
Make this whole class a disposable instead

Reply:
this is the top level class already, we don't have other classes disposing this one. I am not sure how will making it disposable help.

Reply:
Because that saves us having to implement our own Disposable. This is a good practice kind of thing. Yes we could do all this stuff manually - but why bother when VS Code provides that functionality for us in a way that works with other Disposable uses?

Reply:
Just checked the class DisposableStore is not available for extensions.

Reply:
will leave it as is.